typescript/sdk/src/middleware/account/InterchainAccount.ts

```diff
@@ -31,6 +31,12 @@ export class InterchainAccount extends RouterApp<InterchainAccountFactories> {
     super(contractsMap, multiProvider);
   }
 
+  override async remoteChains(chainName: string): Promise<ChainName[]> {
```

Comment:
do you think this change should also be applied for `RouterApp`? I imagine other RouterApps will want to benefit from this. Feels a little weird to have InterchainAccount.remoteChains use the contractsMap but other RouterApps to use what's onchain

Or maybe if reading onchain is necessary to stay the same in RouterApp, there should be a RouterApp fn for reading it onchain and a different one for reading it from the contractsMap?